src/gui/torrentoptionsdialog.cpp

```diff
@@ -0,0 +1,373 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2006  Christophe Dumez <chris@qbittorrent.org>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "torrentoptionsdialog.h"
+
+#include <QMessageBox>
+
+#include "base/bittorrent/infohash.h"
+#include "base/bittorrent/session.h"
+#include "base/bittorrent/torrenthandle.h"
+#include "ui_torrentoptionsdialog.h"
+#include "utils.h"
+
+TorrentOptionsDialog::TorrentOptionsDialog(QWidget *parent, const QVector<BitTorrent::TorrentHandle *> torrents)
+    : QDialog(parent)
+    , m_ui(new Ui::TorrentOptionsDialog)
+{
+    m_ui->setupUi(this);
+
+    const auto *session = BitTorrent::Session::instance();
+    for (const BitTorrent::TorrentHandle *torrent : torrents)
+    {
+        if (torrent)
```

Comment:
You still don't understand me.
You shouldn't check it here (they are valid unless caller side has bugs). And you shouldn't unnecessarily lookup for torrents in constructor. You should store hashes only to use them later (since some torrent can be deleted while dialog is shown).

Reply:
>And you shouldn't unnecessarily lookup for torrents in constructor. You should store hashes only to use them later (since some torrent can be deleted while dialog is shown).

I don't understand what the problem is. I thought that's what i'm doing.

Reply:
>I don't understand what the problem is.

The problem is that you lookup for each torrent by its hash next in this constructor which is unnecessary since you already have valid torrent handles here.

Reply:
Dang! I'm so...